### PR TITLE
New docs folder for developer center

### PR DESCRIPTION
**Pull request by csubira**
The documentation folder (`docs`) need to be restructured in order to meet the requirements of the new developer center.

Four new folders have been added (`guides`, `reference`, `examples` and `support`) filled with redistributed .md files already in the folder.
However, for the moment, since these 'older' files are supplying [carto.com/docs/](www.carto.com/docs/) site, they **cannot be deleted.** 

In this first step, using this branch, the new workflow of documentation repository (for importing and displaying content) is going to be tested.

Once validated, the second step is to get your validation @dgaubert about content (correcting current information or adding new items)

### CODE REVIEW

**Comment by csubira** [comment]
This PR has been updated to:
- Include latest changes of master
- Remove old documentation (used in carto.com/docs, now deprecated and redirected to Developer Center)
- Add some small fixes to make block codes show properly
- Add changes done to documentation in master or in other pending PRs
    - The following PRs should be closed after merging to master: https://github.com/CartoDB/Windshaft-cartodb/pull/942 https://github.com/CartoDB/Windshaft-cartodb/pull/844

The outcome of this changes is already[ in production](https://carto.com/developers/maps-api/), as this branch was the one connected to Developer Center, the only change would me merging this folder to master so it can be updated when developing.

File: docs/guides/02-general-concepts.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## General Concepts
+
+The following concepts are the same for every endpoint in the API except when it's noted explicitly.
+
+### Auth
+
+By default, users do not have access to private tables in CARTO. In order to instantiate a map from private table data an API Key is required. Additionally, to include some endpoints, an API Key must be included (e.g. creating a Named Map).
```
Comment (jgoizueta) [inline docs/guides/02-general-concepts.md]: I'm afraid the second sentence doesn't make much sense to me. What about:

>By default, users do not have access to private tables in CARTO. In order to instantiate a map from private table data an API Key is required. Additionally, an API Key is also required to use some of the API endpoints (e.g. to create a Named Map).

File: docs/MultiLayer-API.md [hunk 2]
```diff
@@ -1,28 +0,0 @@
-The Windshaft-CartoDB MultiLayer API extends the [Windshaft MultiLayer API](https://github.com/CartoDB/Windshaft/blob/master/doc/Multilayer-API.md) in a few ways.
```
Comment (jgoizueta) [inline docs/MultiLayer-API.md]: Why are we removing this rather than keep it somewhere as internal documentation?

**Comment by csubira** [comment]
Apart from the changes suggested in the CR, I changed all the old links to deprecated _carto.com/docs_ site whose redirections were not working properly. Ready to be merged when the checks pass!

**Pull request merged**